.github/workflows/release.yml

```
@@ -4,14 +4,75 @@ on:
   packages: write
 
+env:
+  DRY_RUN: ${{ github.event.inputs.dry_run || 'false' }}
+
 jobs:
+  # ── Dry-run gate ─────────────────────────────────────────────────────────
+  dry-run:
+    name: Dry-Run Gate
+    runs-on: ubuntu-latest
+    outputs:
+      report: ${{ steps.report.outputs.json }}
```

actionlint (1.7.12)

[error] property "json" is not defined in object type {artifact-digest: string; artifact-id: string; artifact-url: string}

(expression)